src/Report_Upload.py

```diff
@@ -33,23 +45,98 @@ def browseFiles():
         
         # Change label contents
         label_file_explorer.configure(text="File Opened: "+filename)
-                                                                                            
-    # Create the root window
-    
+        button_submit = Button(frame,
+                        text = "Submit", command=sub)
+        button_submit.grid(column = 0,row = 5, columnspan=2)
+
     # Set window title
+    def sub():
+        if(repNameBox.get()!="" and patientIDBox.get()!=""):
+            report=repNameBox.get()
+            p_id=patientIDBox.get()
+            # p_id=str(p_id)
+            var1= "Patient ID: "+p_id +"\nName of the report: "+report
+            msg=tk.messagebox.askquestion("Are you sure?", var1 )
+            #if(msg=="yes"):
+            #db connectivity
+            copyReportFile()
+            addReports()
+        else:
+            var2="Please fill in the details"
+            msg1=tk.messagebox.showerror("ERROR", var2)
+
+    def copyReportFile ():
+        # source folder
+        global filename
+        # source filename
+        name = filename[filename.rfind('/')+1 : len(filename)]
+
+        # destination folder
+        newdir = pv.pdfPath
+        # destination filename
+        newfilename = repNameBox.get() + '.pdf'
+
+        newpath = newdir + '\\' + newfilename
+
+        shutil.copy2 ((filename), newdir)
```

Comment:
Wish to understand the newdir=pv.pdfPath

# Python program to explain shutil.copy2() method 
    
# importing shutil module 
import shutil
  
# Source path
source = "/home/User/Documents/file.txt"
  
# Destination path
destination = "/home/User/Documents/file.txt"
  
# Copy the content of
# source to destination
  
try:
    shutil.copy2(source, destination)
    print("File copied successfully.")
  
# If source and destination are same
except shutil.SameFileError:
    print("Source and destination represents the same file.")
  
# If there is any permission issue
except PermissionError:
    print("Permission denied.")
  
# For other errors
except:
    print("Error occurred while copying file.")


We can add this for error handling for shutil.copy2()

Reply:
This is a little long to explain, please remind me to explain this to you in a call.